Update from latest Dev branch fixes

pixie79 opened the pull request.

sublimino committed:
feat: add CLI dev and debug apps

Commit:
migrated to include pxe environment and tests

Commit:
Fixed requirements.yml directory in install.sh

Commit:
feat: GITHUB:7 GITHUB:4 add slack, vagrant and packer roles

Commit:
fix: change deploy order and stop minikube running when root

Commit:
fix: updated slack fix

Commit:
fix: slack deb gub

Commit:
fix: update clamav status check to look for update and updated

Commit:
fix: minikube bash error logging out

Commit:
feat: add yubikey smartcard packages

Commit:
Added instructions for setting up proxy

Commit:
Merge branch 'feat-mpo' of github.com:UKHomeOffice/development_environment into feat-mpo

Commit:
feat: add more yubikey required packages

Commit:
fix: add strace debugging, fix minikube profile and sc support

Commit:
Merge branch 'develop' of https://github.com/sublimino/development_environment into sublimino-develop

* 'develop' of https://github.com/sublimino/development_environment:
  feat: add CLI dev and debug apps

Commit:
Merge branch 'sublimino-develop' into develop

* sublimino-develop:
  feat: add CLI dev and debug apps

Commit:
Update README.md

sublimino committed:
Merge pull request #9 from UKHomeOffice/feat-mpo

Feat mpo

Commit:
fix: add missing pin entry program for gpg

Commit:
fix: remove vmtouch as this does not exist on unbuntu

Commit:
Fix: merging upstream change for clamav

Commit:
Merge pull request #10 from UKHomeOffice/fix-sf

ltgm

Commit:
Fix: corrected syntax in minikube.sh

Commit:
Merge pull request #11 from UKHomeOffice/fix-sf-minikube

Fix: corrected syntax in minikube.sh

Commit:
Fix: updated clamav in requirements.yml

Commit:
Merge pull request #12 from UKHomeOffice/fix-sf-minikube

Fix: updated clamav in requirements.yml

Commit:
Feat: added Sublime role

Commit:
Merge pull request #13 from UKHomeOffice/feat-sf

Feat: added Sublime role

Commit:
fix: upgrade clamav to match upstream

SamFrier committed:
Merge pull request #16 from UKHomeOffice/fix-clamav

fix: upgrade clamav to match upstream

Commit:
fix: stop clamav scan loop bug

SamFrier committed:
Merge pull request #17 from UKHomeOffice/fix-clamav

fix: stop clamav scan loop bug

Commit:
fix: correct grub distro type

Commit:
fix: grub needs a password to prevent menu access

SamFrier committed:
Merge pull request #18 from UKHomeOffice/fix-grub

fix: grub needs a password to prevent menu access

SamFrier committed:
Fixed syntax error

Commit:
feat: update minikube version and fix profile

SamFrier committed:
Merge pull request #19 from UKHomeOffice/fix-minikube

feat: update minikube version and fix profile

Commit:
fix: re add back in kubectl

SamFrier committed:
Merge pull request #20 from UKHomeOffice/fix-minikube

fix: re add back in kubectl

Commit:
fix: after discussion we have decided to remove the buggy profile script and allow users to start minikube themselves as required

Commit:
feat: add gnome control to openvpn

SamFrier committed:
Merge pull request #23 from UKHomeOffice/fix-minikube0812

fix: after discussion we have decided to remove the buggy profile scr…

SamFrier committed:
Merge pull request #22 from UKHomeOffice/feat-openvpn

feat: add gnome control to openvpn

Commit:
feat: force gpg2

SamFrier committed:
Merge pull request #24 from UKHomeOffice/feat-gpg

feat: force gpg2

Commit:
fix: forgot to force creation

SamFrier committed:
Merge pull request #25 from UKHomeOffice/feat-gpg

fix: forgot to force creation

sublimino commented on .vagrant/machines/pxe/virtualbox/synced_folders:
```diff
@@ -0,0 +1 @@
+{"virtualbox":{"/vagrant":{"guestpath":"/vagrant","hostpath":"/home/molliver/GIT/HO/development_environment","disabled":false,"__vagrantfile":true}}}
```
should these be excluded?

Commit:
fix: added debug options to install script

fix: install script now returns the correct exit code when ansible run fails
Fix: corrected syntax in minikube.sh
Fix: updated clamav in requirements.yml
Feat: added Sublime role
fix: upgrade clamav to match upstream
fix: stop clamav scan loop bug
fix: correct grub distro type
fix: grub needs a password to prevent menu access
Fixed syntax error
feat: update minikube version and fix profile
fix: re add back in kubectl
fix: after discussion we have decided to remove the buggy profile script and allow users to start minikube themselves as required
feat: add gnome control to openvpn
feat: force gpg2
fix: forgot to force creation

Commit:
feat: ansible role for sysdig (temporary)

Commit:
feat: added awesome-wm role for devops user (in the same playbook this time)

feat: ansible role for sysdig (temporary)
feat: awesome-wm role in separate playbook for devops users
removed unwanted files

lukebond committed:
Merge pull request #28 from UKHomeOffice/fix-sf

Fix: more debug options for install script

lukebond committed:
Merge pull request #29 from UKHomeOffice/feat-sf

feat: ansible role for sysdig

lukebond commented:
Awaiting @pixie79's return in order to review this. Would like to see commits squashed or it broken up into related commits and PRs.

Commit:
fix: temporarily disabled apt daily updates to prevent conflict with install

Commit:
feat: create ansible meta to allow inclusion as a master role

Commit:
Merge pull request #32 from UKHomeOffice/fix-sf

fix: temporarily disabled apt daily updates to prevent conflict with install

Commit:
fix: remove files that should not have been commited, fix proxy updates in install.sh

Commit:
fix: update proxy logic to work correctly for tests

Commit:
bump: update with upstream

Commit:
Merge branch 'develop' into feat-sf (resolved merge conflicts)

Commit:
fix: renamed awesome-wm role

SamFrier committed:
Merge branch 'develop' into feat-sf

Commit:
Merge pull request #31 from UKHomeOffice/feat-sf

feat: added awesome-wm role for devops user (in the same playbook this time)

Commit:
feat: add openjdk role

Commit:
feat: add FAQ

Commit:
fix: change /boot to have a minimum size of 512M which will allow for kernel upgrades

Commit:
feat: split variables into OS dependencies and GUI or not GUI to allow base use for other repos

SamFrier committed:
Merge pull request #37 from UKHomeOffice/feat-gui

feat: split variables into OS dependencies and GUI

SamFrier committed:
Merge pull request #35 from UKHomeOffice/feat-openjdk

feat: add openjdk role

Commit:
fix: update proxy settings

SamFrier committed:
Merge pull request #38 from UKHomeOffice/fix-proxy

fix: update proxy settings

Commit:
fix: typo in os_packages causes ansible run to fail

fix: line in box setup script is now properly commented out

Commit:
feat: added openjdk to site.yml

Commit:
Merge pull request #42 from UKHomeOffice/feat-openjdk

feat: added openjdk to site.yml

Commit:
Merge pull request #40 from UKHomeOffice/fix-typo

fix: typo in os_packages caused ansible run to fail

Commit:
fix: systemctl stops apt-daily.service instead of disabling timer

Commit:
Merge pull request #48 from UKHomeOffice/fix-apt

fix: systemctl stops apt-daily.service instead of disabling timer

SamFrier committed:
feat: updated FAQ

SamFrier committed:
feat: extended FAQ

SamFrier committed:
fix: syntax error in FAQ

SamFrier committed:
fix: removed outdated section of readme

Commit:
fix: rkhunter update no longer fails the ansible run if no updates needed

Commit:
Merge pull request #54 from UKHomeOffice/fix-rkhunter-update

fix: rkhunter update no longer fails the ansible run if no updates needed

Commit:
feat: eclipse role

fix: using a UK download mirror

Commit:
Merge pull request #56 from UKHomeOffice/feat-eclipse

feat: eclipse role

Commit:
fix: eclipse now has application shortcut

SamFrier committed:
Updated readme with instructions for installing awesome-wm

SamFrier committed:
Fix syntax error in readme

Commit:
fix: temporarily removed eclipse role

freddyp92 committed:
added configuration for default keyboard language (uk)

Ian-T-Price committed:
Merge pull request #60 from UKHomeOffice/fix-keyboard

added configuration for default keyboard language (uk)

Commit:
Merge pull request #59 from UKHomeOffice/feat-eclipse

fix: temporarily removed eclipse role
